src/validators.js

```diff
@@ -133,8 +134,16 @@ function validateHasValueProperty(context, focusNode, valueNode, constraint) {
 function validateIn(context, focusNode, valueNode, constraint) {
   const { sh } = context.ns
   const inNode = constraint.getParameterValue(sh.in)
+  if (!constraint.nodeSets) {
+    constraint.nodeSets = new TermMap()
+  }
+  let nodeSet = constraint.nodeSets.get(inNode)
+  if (!nodeSet) {
+    nodeSet = new NodeSet(rdfListToArray(context.$shapes.node(inNode)))
+    constraint.nodeSets.set(inNode, nodeSet)
+  }
```

Comment:
Do we need a list of node sets? I would expect a constraint object to represent a single `sh:in` constraint

```suggestion
  if (!constraint.nodeSet) {
    const inNode = constraint.getParameterValue(sh.in)
    constraint.nodeSet = new NodeSet(rdfListToArray(context.$shapes.node(inNode)))
    constraint.nodeSet = new TermMap()
  }
  const { nodeSet } = constraint
```

Reply:
my understanding is that all the shapes with a `sh:in` constraint are sharing the same instance

Reply:
I don't think so. `getParameterValue` gets the `sh:in` from the shape pointer.

Reply:
correct